src/lib/components/notifications/index.ts

```diff
@@ -1,30 +1,43 @@
  import { writable, derived } from "svelte/store";
  import type { Readable } from "svelte/store";
+import type { Socket } from "socket.io";
  
  const NOTIFICATION_TIMEOUT = 2000;
  
  export interface NotificationStore extends Readable<object> {
    timeoutID: ReturnType<typeof setTimeout>;
-   send: Function;
-   clear: Function;
-   listenToSocket: Function;
+   send: (args:NotificationMssageArguments) => void;
+   clear: () => void;
+   listenToSocket: (s:Socket) => void;
  }
  
+interface NotificationMessageArguments {
+  message: string;
+  type?: string;
+  options?: Options;
+}
+
  interface NotificationMessage {
    id: string;
-   type: string;
+   type?: string;
    message: string;
+   options?: Options;
+ }
+
+ interface Options {
+   width?: number;
  }
  
  export function createNotificationStore(): NotificationStore {
    const _notification = writable({ id: undefined });
    let timeout: ReturnType<typeof setTimeout>;
  
-   function send({ message, type = "default" }): void {
+   function send({ message, type = "default", options = {} }: NotificationMessageArguments): void {
```

Comment:
just a note that the changes to the notifications are largey to support a new `options` argument, which enables the user to pass in things like a `width` param.